docs/guides/custom_artifact_definition.md

```diff
@@ -0,0 +1,256 @@
+              }
+            },
+            "bar": {
+              "title": "Bar",
+              "description": "Bar description",
+              "additionalProperties": false,
+              "examples": [
+              ],
+              "type": "string"
+            }
+          }
+        },
+        "authentication": {
+          "title": "Authentication configuration",
+          "type": "object",
+          "required": [
+            "foobar"
+          ],
+          "properties": {
+            "foobar": {
+              "title": "Foobar",
+              "type": "string"
+            }
+          }
+        },
+        "security": {
+          "title": "Security",
+          "description": "Security Configuration",
+          "additionalProperties": false,
+          "required": [],
+          "properties": {
+            "iam": {
+              "title": "IAM",
+              "description": "IAM Roles And Scopes",
+              "additionalProperties": false,
+              "patternProperties": {
+                "^[a-z]+[a-z_]*[a-z]$": {
+                  "type": "object",
+                  "required": [
+                    "role",
+                    "scope"
+                  ],
+                  "properties": {
+                    "role": {
+                      "title": "Role",
+                      "description": "Cloud Role",
+                      "pattern": "^[a-zA-Z ]+$",
+                      "message": {
+                        "pattern": "Must be a valid Cloud Role (uppercase, lowercase letters and spaces)"
+                      },
+                      "examples": [
+                        "Data Reader"
+                      ]
+                    },
+                    "scope": {
+                      "title": "Scope",
+                      "description": "Cloud IAM Scope (cloud resource identifier)",
+                      "type": "string"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "specs": {
+      "title": "Artifact Specs",
+      "type": "object",
+      "properties": {
+        "cloud": {
+          "properties": {
+            "region": {
+              "type": "string",
+              "title": "Cloud Region",
+              "description": "Select the cloud region you'd like to provision your resources in."
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Step 5: Publishing to Massdriver
+
+Got your definition looking sharp? Use the `mass definition publish -f /path/to/definition.json` command in the CLI to send it out into the world. Remember to set your `$md.access` field to `private` to keep it just between you and your team.
+
+### Step 6: Fetching Your Masterpiece
+
+Once published, snag your artifact definition with the `mass definition get org/definition-name` command to confirm it's ready for action in your bundles.
+
+### Step 7: Using Your Custom Artifact Definition
```

Comment:
We'll need to add something about doing this via the `jq` methods that works across all provisioners. But no need to hold this PR up.